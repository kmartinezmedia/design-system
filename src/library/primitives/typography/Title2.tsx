import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Text, TextProps } from './Text';
import {
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
} from '@designSystem/theme';

export const Title2: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 5 },
  ...textProps
}) => (
  <Text style={styles.Title2} spacing={spacing} {...textProps}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  Title2: {
    fontSize: fontSize.title2,
    lineHeight: lineHeight.title2,
    fontFamily: fontFamily.regular,
    fontWeight: fontWeight.regular,
  },
});
