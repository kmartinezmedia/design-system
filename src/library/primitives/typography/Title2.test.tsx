import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Title2 } from './Title2';

describe('Title2', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Title2>Hey, lil 🐛</Title2>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
