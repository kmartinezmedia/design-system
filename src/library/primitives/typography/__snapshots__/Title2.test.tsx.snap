// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title2 default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Regular",
        "fontSize": 28,
        "fontWeight": "400",
        "lineHeight": 40,
      },
      Object {
        "paddingBottom": 20,
      },
      Object {
        "color": "#071422",
        "lineHeight": 40,
        "overflow": "visible",
        "textAlign": undefined,
      },
      undefined,
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
